src/python/pants/backend/python/lint/ruff/rules_integration_test.py

```diff
@@ -0,0 +1,163 @@
+    )
+    assert result.stderr == ""
+    assert result.stdout == ""
+    assert not result.did_change
+    assert result.output == get_snapshot(rule_runner, {"f.py": GOOD_FILE})
+
+
+def test_failing(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"f.py": BAD_FILE, "BUILD": "python_sources(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    fix_result = run_ruff(rule_runner, [tgt])
+    assert fix_result.stdout == "Found 1 error(s) (1 fixed, 0 remaining).\n"
+    assert fix_result.stderr == ""
+    assert fix_result.did_change
+    assert fix_result.output == get_snapshot(rule_runner, {"f.py": GOOD_FILE})
+
+
+def test_multiple_targets(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "good.py": GOOD_FILE,
+            "bad.py": BAD_FILE,
+            "BUILD": "python_sources(name='t')",
+        }
+    )
+    tgts = [
+        rule_runner.get_target(Address("", target_name="t", relative_file_path="good.py")),
+        rule_runner.get_target(Address("", target_name="t", relative_file_path="bad.py")),
+    ]
+    fix_result = run_ruff(rule_runner, tgts)
+    assert fix_result.output == get_snapshot(
+        rule_runner, {"good.py": GOOD_FILE, "bad.py": GOOD_FILE}
+    )
+    assert fix_result.did_change is True
+
+
+@pytest.mark.parametrize(
+    "file_path,config_path,extra_args,should_change",
+    (
+        [Path("f.py"), Path("pyproject.toml"), [], False],
+        [Path("f.py"), Path("ruff.toml"), [], False],
+        [Path("custom/f.py"), Path("custom/ruff.toml"), [], False],
+        [Path("f.py"), Path("custom/ruff.toml"), ["--ruff-config=custom/ruff.toml"], False],
+        [Path("f.py"), Path("custom/ruff.toml"), [], True],
+    ),
+)
+def test_config_file(
+    rule_runner: RuleRunner,
+    file_path: Path,
+    config_path: Path,
+    extra_args: list[str],
+    should_change: bool,
+) -> None:
+    hierarchy = "[tool.ruff]\n" if config_path.stem == "pyproject" else ""
+    rule_runner.write_files(
+        {
+            file_path: BAD_FILE,
+            file_path.parent / "BUILD": "python_sources()",
+            config_path: f'{hierarchy}ignore = ["F541"]',
+        }
+    )
+    spec_path = str(file_path.parent).replace(".", "")
+    rel_file_path = file_path.relative_to(*file_path.parts[:1]) if spec_path else file_path
+    addr = Address(spec_path, relative_file_path=str(rel_file_path))
+    tgt = rule_runner.get_target(addr)
+    fix_result = run_ruff(rule_runner, [tgt], extra_args=extra_args)
+    assert fix_result.did_change is should_change
+
+
+def test_skip(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"f.py": BAD_FILE, "BUILD": "python_sources(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    result = run_ruff(rule_runner, [tgt], extra_args=["--ruff-skip"])
+    assert not result
```

Comment:
I couldn't get the skip argument to work for some reason, I would appreciate some help with this

Reply:
Is the skip rule registered on the test runner?

Reply:
Do you mean like this?
```diff
@pytest.fixture
def rule_runner() -> RuleRunner:
    return RuleRunner(
        rules=[
            *ruff_rules(),
+            *skip_field.rules(),
            *ruff_subsystem_rules(),
            *config_files.rules(),
            *target_types_rules.rules(),
            QueryRule(FmtResult, [RuffRequest.Batch]),
            QueryRule(SourceFiles, (SourceFilesRequest,)),
        ],
        target_types=[PythonSourcesGeneratorTarget],
    )
```
```python
# skip_field.py
....
def rules():
    return [
        PythonSourcesGeneratorTarget.register_plugin_field(SkipRuffField),
        PythonSourceTarget.register_plugin_field(SkipRuffField),
        PythonTestsGeneratorTarget.register_plugin_field(SkipRuffField),
        PythonTestTarget.register_plugin_field(SkipRuffField),
        PythonTestUtilsGeneratorTarget.register_plugin_field(SkipRuffField),
    ]
```
It didn't work

Reply:
FYI There's a difference between the `skip` field on a target (such as your code snippet) and the skip argument on goals.

Reply:
So this test won't ever pass because the skip handling is done in the partitioner rule, and `run_ruff` only requests the `FixResult`.


If you look at other formatters, if they opt into the default partitioner they don't test `--skip`, so feel free to just delete this test :smile:

Reply:
Thanks, removed :)